api/v1alpha1/controlplane_types.go

```diff
@@ -22,10 +22,17 @@ import (
 
 // ControlPlaneSpec defines the desired state of ControlPlane
 type ControlPlaneSpec struct {
-	Type               ControlPlaneType  `json:"type,omitempty"`
-	Backend            BackendDBType     `json:"backend,omitempty"`
-	PostCreateHook     *string           `json:"postCreateHook,omitempty"`
-	PostCreateHookVars map[string]string `json:"postCreateHookVars,omitempty"`
+	Type    ControlPlaneType `json:"type,omitempty"`
+	Backend BackendDBType    `json:"backend,omitempty"`
+	// BootstrapSecretRef contains a reference to the kubeconfig used to bootstrap adoption of
+	// an external cluster
+	// +optional
+	BootstrapSecretRef *SecretReference `json:"bootstrapSecretRef,omitempty"`
+	// expiration time for token of adopted cluster
+	// +optional
+	AdoptedTokenExpirationSeconds *int64            `json:"adoptedTokenExpirationSeconds,omitempty"`
```

Comment:
(A) If the type is "external", shouldn't this field's name start with "External" rather than "Adopted"?

(B) Is this parameter really only relevant to the case of external type? Isn't this a question that arises for every type (possibly excepting host)?